filter/series_by_tag.go

```diff
@@ -163,13 +170,26 @@ func createMatchingHandlerForOneTag(spec TagSpec) MatchingHandler {
 			return value != spec.Value
 		}
 	case MatchOperator:
-		allowMatchEmpty = true
-		matchRegex := regexp.MustCompile("^" + spec.Value)
+		value := cleanAsterisks(spec.Value)
+		if !strings.HasPrefix(value, "^") {
+			value = ".*" + value
+		}
+		if !strings.HasSuffix(value, "$") {
+			value += ".*"
+		}
+		matchRegex := regexp.MustCompile(value)
 		matchingHandlerCondition = func(value string) bool {
 			return matchRegex.MatchString(value)
 		}
 	case NotMatchOperator:
-		matchRegex := regexp.MustCompile("^" + spec.Value)
+		value := cleanAsterisks(spec.Value)
```

Comment:
Why is needed ? Replace `*` to `.*` needed only if full string is `*`. Also may be add more tests for `*`

Reply:
@msaf1980 I don't know if i missed any tests, I have tried the following special cases here: https://github.com/moira-alert/moira/pull/896/files#diff-4e74ec0972e1312934727b9a6b03635e81344c8d854205ea594290ad604bedc7

- empty
- single asterisks
- asterisk after an asterisk
- properly written asterisk should remain i.e. 
    - `.*` => `..*` ❌
    - `.*` => `.*` ✅
- when regexp has been anchored with `^` and `$`
- prefix and suffix asterisks
- mixed (correct and incorrectly written) asterisks

Reply:
In graphite-clickhouse seriesByTags `*` replaced to `.*` only in wildcards (it's not regular expression, but translated to regexp for check). One case whan `*` translated to '.*` in regexp, is a bad regexp `*`.

Reply:
Also, add tests for regexp like `a*b` and `a.*b` for check why replacement `*` to `.*` is a bad thing.

Reply:
Done